src/message.rs

```diff
@@ -18,6 +20,23 @@ pub(crate) struct MessageFilterFailure {
     pub(crate) context: &'static str,
 }
 
+pub(crate) fn clean_mentions(content: &str, mentions: &[Mention]) -> String {
```

Comment:
This has the effect of _always_ copying the content string, which is up to 4096 bytes (or more, depending on the exact details of Discord's character limit). That could be kind of expensive if the message is very long but doesn't include any mentions.

I'd like to either:
- avoid invoking this method if we have no mentions, OR
- return a `Cow<str>` or something in the case of there being no mentions at all